src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipePayloadDecoder.cs

```diff
@@ -110,16 +110,29 @@ internal static object[] DecodePayload(ref EventSource.EventMetadata metadata, R
                 }
                 else if (parameterType == typeof(string))
                 {
-                    ReadOnlySpan<char> charPayload = MemoryMarshal.Cast<byte, char>(payload);
-                    int charCount = charPayload.IndexOf('\0');
-                    if (charCount < 0)
+                    // Try to find null terminator (0x00) from the byte span
+                    // NOTE: we do this by hand instead of using payload.IndexOf('\n') because payload may be unaligned due to 
```

Comment:
Nit: This comment talks about `payload.IndexOf('\n')`, but the code is looking for zeros (`\n` is not a zero).